contributors/design-proposals/local-storage-overview.md

```diff
@@ -302,22 +312,25 @@ Since local PVs are only accessible from specific nodes, a new PV-node associati
 5. If a pod dies and is replaced by a new one that reuses existing PVCs, the pod will be placed on the same node where the corresponding PVs exist. Stateful Pods are expected to have a high enough priority which will result in such pods preempting other low priority pods if necessary to run on a specific node.
 6. Forgiveness policies can be specified as tolerations in the pod spec for each failure scenario.  No toleration specified means that the failure is not tolerated.  In that case, the PVC will immediately be unbound, and the pod will be rescheduled to obtain a new PV.  If a toleration is set, by default, it will be tolerated forever.  `tolerationSeconds` can be specified to allow for a timeout period before the PVC gets unbound.
 
-  Node taints already exist today.  New PV and scheduling taints can be added to handle additional failure use cases when using local storage.  A new PV taint will be introduced to handle unhealthy volumes.  The addon or another external entity can monitor the volumes and add a taint when it detects that it is unhealthy.  A scheduling taint could signal a scheduling failure for the pod due to being unable to fit on the node.
+  Node taints already exist today.  A new PV taint will be introduced to handle unhealthy volumes.  The addon or another external entity can monitor the volumes and add a taint when it detects that it is unhealthy.  Pod scheduling failures are specified separately as a timeout.
   ```yaml
-  nodeTolerations:
-    - key: node.alpha.kubernetes.io/notReady
-      operator: TolerationOpExists
-      tolerationSeconds: 600
-    - key: node.alpha.kubernetes.io/unreachable
-      operator: TolerationOpExists
-      tolerationSeconds: 1200
-  pvTolerations:
-    - key: storage.kubernetes.io/pvUnhealthy
-      operator: TolerationOpExists
-  schedulingTolerations:
-    - key: scheduler.kubernetes.io/podCannotFit
-      operator: TolerationOpExists
-      tolerationSeconds: 600
+  apiVersion: v1
+  kind: pod
+  metadata:
+    name: foo
+  spec:
+    <snip>
+    nodeTolerations:
+      - key: node.alpha.kubernetes.io/notReady
+        operator: TolerationOpExists
+        tolerationSeconds: 600
+      - key: node.alpha.kubernetes.io/unreachable
+        operator: TolerationOpExists
+        tolerationSeconds: 1200
+    pvTolerations:
+      - key: storage.kubernetes.io/pvUnhealthy
+        operator: TolerationOpExists
+    schedulingFailureTimeoutSeconds: 600
```

Comment:
This feature can only be supported on a persistent volume claim and so it has to be a feature in the persistentvolumeclaimsource object.